cmd/kube-apiserver/app/testing/testserver.go

```diff
@@ -87,7 +87,8 @@ type TestServerInstanceOptions struct {
 	// Useful for testing server configurations expected to prevent /healthz from completing.
 	SkipHealthzCheck bool
 	// Enable cert-auth for the kube-apiserver
-	EnableCertAuth bool
+	EnableCertAuth                  bool
+	RequestHeaderUIDHeadersOverride []string
```

Comment:
Instead of wiring a new override config, lets merge the default requester header options with whatever is specified via flags (with the flags having precedence).